synapse/api/filtering.py

```diff
@@ -253,19 +254,30 @@ def check(self, event):
         Returns:
             bool: True if the event matches
         """
-        sender = event.get("sender", None)
-        if not sender:
-            # Presence events have their 'sender' in content.user_id
-            content = event.get("content")
-            # account_data has been allowed to have non-dict content, so check type first
-            if isinstance(content, dict):
-                sender = content.get("user_id")
+        if isinstance(event, UserPresenceState):
```

Comment:
Could you comment what's going on here? Just a quick comment to explain why both branches are necessary would be enough.